src/mission_start.cpp

```diff
@@ -735,15 +729,44 @@ void mission_start::reveal_lab_train_depot( mission *miss )
     }
 
     if( !comppoint ) {
-        debugmsg( "Could not find a computer in the lab train depot, mission will fail." );
-        return;
+        return false;
     }
 
     computer *tmpcomp = compmap.computer_at( *comppoint );
-    tmpcomp->set_mission( miss->get_id() );
+    tmpcomp->set_mission( mission_id );
     tmpcomp->add_option( _( "Download Routing Software" ), COMPACT_DOWNLOAD_SOFTWARE, 0 );
 
     compmap.save();
+    return true;
+}
+
+void mission_start::reveal_lab_train_depot( mission *miss )
+{
+    Character &player_character = get_player_character();
+    // Find and prepare lab location.
+    tripoint_abs_omt loc = player_character.global_omt_location();
+    loc.z() = -4;  // tunnels are at z = -4
+    tripoint_abs_omt place;
+    const int mission_id = miss->get_id();
+
+    omt_find_params params;
+    std::pair<std::string, ot_match_type> temp_pair;
+    temp_pair.first = "lab_train_depot";
+    temp_pair.second = ot_match_type::type;
+    params.types.push_back( temp_pair );
+
+    const std::vector<tripoint_abs_omt> all_omts_near = overmap_buffer.find_all( loc, params );
+    // sort it by range
+    std::multimap<int, tripoint_abs_omt> omts_by_range;
+    for( const tripoint_abs_omt &location : all_omts_near ) {
+        omts_by_range.emplace( rl_dist( loc, location ), location );
+    }
+    for( const std::pair<int, const tripoint_abs_omt> &location : omts_by_range ) {
```

Comment:
This is the right way to iterate through a `multimap` in ranged based for loop if not using `auto`
```suggestion
    for( const std::pair<const int, tripoint_abs_omt> &location : omts_by_range ) {
```